bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -405,7 +405,27 @@ public boolean resumeSuspendedClients(final int numberOfInMemBlockPermits)
 
   public boolean isMemoryBlockAvailable()
   {
-    return (storage == null) || (numberOfInMemBlockPermits.get() > 0);
+    return (numberOfInMemBlockPermits.get() > 0) || ((storage == null) && !areSubscribersBehindByMax());
```

Comment:
That's the plan for spooling enabled case as well in APEXCORE-646 mentioned briefly in the description. Will make it a subtask there and do the parameter at that time.